pilot/pkg/networking/core/v1alpha3/route/route.go

```diff
@@ -418,8 +418,27 @@ func translateRoute(push *model.PushContext, node *model.Proxy, in *networking.H
 		out.ResponseHeadersToRemove = responseHeadersToRemove
 
 		if in.Mirror != nil {
-			n := GetDestinationCluster(in.Mirror, serviceRegistry[host.Name(in.Mirror.Host)], port)
-			action.RequestMirrorPolicy = &route.RouteAction_RequestMirrorPolicy{Cluster: n}
+			var percent uint32 = 100
+
+			if in.MirrorPercent != nil {
+				percent = in.MirrorPercent.GetValue()
+				if percent > 100 {
+					percent = 100
+				}
```

Comment:
this should be done in validation.

Reply:
as well

Reply:
this is already being done there.

Reply:
+1. This condition could be dropped altogether since the input has been validated already.

Reply:
removed.